src/matcha_ml/cli/_validation.py

```diff
@@ -225,20 +225,28 @@ def check_current_deployment_exists() -> bool:
     with open(MATCHA_STATE_DIR) as f:
         data = json.load(f)
 
-    resource_group_name = data["cloud"]["resource-group-name"]
+    # Check if resource-group-name is present in matcha.state file
+    if "cloud" in data:
```

Comment:
Thinking about this, wouldn't it make more sense to have a single way of getting this information irrespective of whether `provision` fails? It'd make this a little cleaner

Reply:
Based on what is stored inside state file, there are 2 ways of getting resource group name.

1. When provision fails midway, since we are adding location and prefix we get it using `data['prefix']` approach.
2. When provision is success, the provisioned resources would have overwritten the location and prefix, contain whatever is output from terraform output as is, basically what is was before. For this approach, we use d`ata['cloud']` approach.

As noted in above comment, we can also include location and prefix inside state file instead of overwriting which is the current behaviour of the implementation.